### PR TITLE
feat(sessions): add dedup command for overlapping session records

### PR DESCRIPTION
**Pull request by TimeToBuildBob**
## What

Adds a `dedup` subcommand to `gptme-sessions` — the second half of the
auto-tagging-and-dedup task (auto-tag landed in #1028).

Merges duplicate session records into the richest record:
- Two records are duplicates when they share a **harness** and their time
  intervals overlap by at least `--min-overlap` (default `0.6`), measured as
  **Jaccard** overlap (intersection / union).
- The richest record per group is kept and **enriched** with any metadata it
  was missing; the duplicates are tagged with `duplicate_of` so analytics can
  filter them.
- Records are **never deleted** — trajectory/session data is preserved (the
  trajectory-persistence invariant). Re-running is idempotent: already-merged
  duplicates are skipped.
- `--dry-run` previews the merge plan without rewriting the store.

## Why Jaccard, not fraction-of-shorter

A short session nested inside a longer, genuinely *different* session scores
1.0 under "fraction of the shorter interval", producing false duplicates. On
the real store that over-flagged ~1634 records; Jaccard (intersection/union)
only flags near-identical intervals — two recordings of the *same* session —
cutting it to ~426 at the 0.6 default (~76 at 0.9).

## Tests

`TestDedupCommand` covers: dry-run non-persistence, richer-record merge +
`duplicate_of` tagging + deliverable union, idempotency, cross-harness
isolation, and the `--min-overlap` threshold boundary.

- `uv run pytest packages/gptme-sessions/tests/test_cli.py -q` → 68 passed
- `make -C packages/gptme-sessions typecheck` → clean

## Note

This PR ships the *capability* and validates it via `--dry-run` on the real
store. It does **not** run the destructive merge on the canonical
`session-records.jsonl` (used for grading/bandit) — that is a deliberate
operational decision left for a reviewed run.

### CODE REVIEW

**Comment by TimeToBuildBob** [comment]
Fixed in 1ba6944 — addressed both issues Greptile flagged:

- **Zero-score overwrite**: Removed `0` from `_EMPTY_VALUES`. Python's `0 == 0.0` equality caused a keeper's legitimate `llm_judge_score`/`trajectory_grade` of `0.0` to be treated as empty and overwritten by the duplicate's value. The correct sentinel for "not set" on numeric fields is `None`.

- **Unreachable zero-duration path**: Reordered `_overlap_ratio` to compute `union` before `intersection`. The old code returned `0.0` on `intersection <= 0` first, so the `union <= 0` branch (identical zero-length instants) was dead code. Now zero-duration identical records correctly return `1.0`.

Tests added for both edge cases (`test_dedup_preserves_zero_score`, `test_dedup_identical_zero_duration`). All 7 dedup tests pass.

**Comment by TimeToBuildBob** [comment]
@greptileai review

File: packages/gptme-sessions/src/gptme_sessions/cli.py [hunk 1]
```diff
@@ -388,6 +388,168 @@ def _match_auto_tag_override(
...
+        if other not in _EMPTY_VALUES:
+            if apply:
+                setattr(keeper, name, other)
+            filled.append(name)
+    existing = set(keeper.deliverables or [])
+    extra = [d for d in (duplicate.deliverables or []) if d not in existing]
+    if extra:
+        if apply:
+            keeper.deliverables.extend(extra)
+        filled.append("deliverables")
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply packages/gptme-sessions/src/gptme_sessions/cli.py:553]: Fixed in e02ffbb — guard added before : if  we assign  instead of calling  on None. Test added:  exercises the crash path directly.

File: packages/gptme-sessions/src/gptme_sessions/cli.py [hunk 2]
```diff
@@ -388,6 +388,166 @@ def _match_auto_tag_override(
...
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply packages/gptme-sessions/src/gptme_sessions/cli.py:553]: Fixed in e02ffbb — both are now merged in `_merge_into_keeper`: `deliverable_details` via value-equality union (same None-guard), `grades` by adding keys the keeper lacks. Test added: `test_dedup_merges_grades_and_deliverable_details` verifies both.

File: packages/gptme-sessions/tests/test_cli.py [hunk 3]
```diff
@@ -1055,3 +1055,133 @@ def test_auto_tag_mapping_file_overrides_signal_category(
...
+
+        rc, out = _invoke(["dedup"], tmp_path)
+
+        assert rc == 0
+        assert "No duplicate session records found" in out
+
+    def test_dedup_respects_min_overlap_threshold(self, tmp_path: Path) -> None:
+        """Records overlapping below --min-overlap are left untouched."""
+        store = SessionStore(sessions_dir=tmp_path)
+        # 2-min overlap of a 10-min interval = 0.2, below the 0.6 default.
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply packages/gptme-sessions/tests/test_cli.py]: Fixed in e02ffbb — comment now reads: `Jaccard = intersection/union = 2 min / 18 min ≈ 0.11, below the 0.6 default.`

**Comment by TimeToBuildBob** [comment]
@greptileai review
